docs/reference/setup/install/docker.asciidoc

```diff
@@ -0,0 +1,290 @@
+
+[[docker-prod-cluster-composefile]]
+`docker-compose.yml`:
+ifeval::["{release-state}"=="unreleased"]
+
+Version {version} of the Elasticsearch Docker image has not yet been released, so a `docker-compose.yml` is not available for this version.
+
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+["source","yaml",subs="attributes"]
+--------------------------------------------
+version: '2'
+services:
+  elasticsearch1:
+    image: docker.elastic.co/elasticsearch/elasticsearch:{version}
+    container_name: elasticsearch1
+    environment:
+      - cluster.name=docker-cluster
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    mem_limit: 1g
+    cap_add:
+      - IPC_LOCK
+    volumes:
+      - esdata1:/usr/share/elasticsearch/data
+    ports:
+      - 9200:9200
+    networks:
+      - esnet
+  elasticsearch2:
+    image: docker.elastic.co/elasticsearch/elasticsearch:{version}
+    environment:
+      - cluster.name=docker-cluster
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+      - "discovery.zen.ping.unicast.hosts=elasticsearch1"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+      nofile:
+        soft: 65536
+        hard: 65536
+    mem_limit: 1g
+    cap_add:
+      - IPC_LOCK
+    volumes:
+      - esdata2:/usr/share/elasticsearch/data
+    networks:
+      - esnet
+
+volumes:
+  esdata1:
+    driver: local
+  esdata2:
+    driver: local
+
+networks:
+  esnet:
+    driver: bridge
+--------------------------------------------
+endif::[]
+
+To stop the cluster, type `docker-compose down`. Data volumes will persist, so it's possible to start the cluster again with the same data using `docker-compose up`.
+To destroy the cluster **and the data volumes** just type `docker-compose down -v`.
+
+===== Inspect status of cluster:
+
+["source","sh"]
+--------------------------------------------
+curl -u elastic http://127.0.0.1:9200/_cat/health
+Enter host password for user 'elastic':
+1472225929 15:38:49 docker-cluster green 2 2 4 2 0 0 0 0 - 100.0%
+--------------------------------------------
+
+Log messages go to the console and are handled by the configured Docker logging driver. By default you can access logs with `docker logs`.
+
+[[docker-configuration-methods]]
+==== Configuring Elasticsearch with Docker
+
+Elasticsearch loads its configuration from files under `/usr/share/elasticsearch/config/`. These configuration files are documented in <<settings>> and <<es-java-opts>>.
+
+The image offers several methods for configuring Elasticsearch settings.
+
+The conventional approach is to provide customized files, i.e. `elasticsearch.yml`, but it's also possible to use environment variables to set options.
+
+===== A. Present the parameters via Docker environment variables
```

Comment:
I'd present these options as a definition list instead of headers. You'll need to use `--` blocks because the "definitions" contain mixed markup, eg:

    Present the parameters via Docker environment variables::
    +
    --
    For example, to define the cluster name with `docker run` you can pass `-e "cluster.name=mynewclustername"`. Double quotes are required.

    NOTE: There is a difference between defining <<_setting_default_settings,default settings>> and normal settings. The former are prefixed with `default.` and can not override normal settings, if defined.
    --

    Bind-mounted configuration::
    +
    --
    etc...
    --